drivers/storage/portworx/component/prometheus.go

```diff
@@ -625,14 +625,36 @@ func (c *prometheus) createPrometheusInstance(
 			RuleSelector: &metav1.LabelSelector{
 				MatchLabels: prometheusRuleLabels(),
 			},
-			Resources: v1.ResourceRequirements{
-				Requests: map[v1.ResourceName]resource.Quantity{
-					v1.ResourceMemory: resource.MustParse("400Mi"),
-				},
-			},
 		},
 	}
 
+	if cluster.Spec.Monitoring.Prometheus.Resources.Limits != nil {
+		prometheusInst.Spec.Resources.Limits = cluster.Spec.Monitoring.Prometheus.Resources.Limits
+	} else {
+		prometheusInst.Spec.Resources.Limits = map[v1.ResourceName]resource.Quantity{
+			v1.ResourceMemory:           resource.MustParse("800Mi"),
```

Comment:
but I feel pod consuming more memory is better than OOM crashing in those scenarios.

Reply:
it may impact other pods running on the worker node. for example metrics collector had complaint about memory usage and later we added default limits too.